.github/workflows/autorelease-rubygem.yml

```diff
@@ -0,0 +1,96 @@
+
+jobs:
+  release:
+    name: Release Gem
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        show-progress: false
+    - uses: ruby/setup-ruby@v1
+      with:
+        rubygems: latest
+        bundler-cache: true
+    - name: "Determine HEAD of default branch"
+      id: fetch_default_branch_head
+      run: echo "sha=$(git ls-remote origin HEAD | cut -f 1)" >> "$GITHUB_OUTPUT"
+    - if: github.sha == steps.fetch_default_branch_head.outputs.sha
+      name: "Auto-release rubygem"
+      env:
+        GEM_NAME: ${{ inputs.gem_name }}
+        GH_TOKEN: ${{ github.token }}
+      run: |
+        set -euo pipefail
+
+        CURR_VERSION=$(ruby -e "puts eval(File.read('$GEM_NAME.gemspec')).version")
+        NEXT_VERSION=$(echo $CURR_VERSION | cut -d . -f1).$(echo $CURR_VERSION | cut -d . -f2).$(( $(echo $CURR_VERSION | cut -d . -f3) + 1 ))
+        BRANCH="release-$NEXT_VERSION"
+        PR_TITLE="Release v$NEXT_VERSION"
+        PR_DESCRIPTION="This is an automated PR to bump the version number to $NEXT_VERSION. This is a patch-level bump, as the only changes since v$CURR_VERSION are dependency updates made by Dependabot."
+
+        if ! [ -f CHANGELOG.md ]; then
+          echo "CHANGELOG.md does not exist. Exiting."
+          exit 0
+        fi
+
+        if ! grep -qR --include "version.rb" $CURR_VERSION; then
+          echo "Can't find a version.rb containing the string '$CURR_VERSION'. Exiting."
+          exit 0
+        fi
+
+        if git ls-remote --exit-code origin refs/heads/$BRANCH > /dev/null; then
+          echo "Branch 'release-$NEXT_VERSION' already exists on the remote. Exiting."
+          exit 0
+        fi
+
+        if [ "$(git log v$CURR_VERSION..main --no-merges --format=format:%an | sort | uniq)" != "dependabot[bot]" ]; then
+          echo "Gem has no unreleased changes, or those changes include commits from users other than Dependabot. Exiting."
+          exit 0
+        fi
+
+        find . -type f -iname version.rb -exec sed -i "s/$CURR_VERSION/$NEXT_VERSION/g" {} \;
+
+        printf "# v$NEXT_VERSION\n\n* Update dependencies\n\n" | cat - CHANGELOG.md > NEW_CHANGELOG.md
+        mv NEW_CHANGELOG.md CHANGELOG.md
+
+        git config --global user.name "GOV.UK Continuous Integration User"
+        git config --global user.email "govuk-ci@users.noreply.github.com"
+
+        git checkout -b $BRANCH
+        git add .
+        git commit -m "$PR_TITLE"
+        git push origin $BRANCH
+
+        gh pr create -H $BRANCH --title "$PR_TITLE" --body "$PR_DESCRIPTION"
+    - if: github.sha != steps.fetch_default_branch_head.outputs.sha
+      name: "Skip release: commit is not HEAD of default branch"
+      run: |
+        echo "Skipping release because commit (${GITHUB_SHA}) isn't the HEAD of the default branch (${{ steps.fetch_default_branch_head.outputs.sha }})"
```

Comment:
```suggestion
        echo "Skipping release because commit (${github.sha}) isn't the HEAD of the default branch (${{ steps.fetch_default_branch_head.outputs.sha }})"
```

Reply:
I think this would need double squiggly brackets (e.g. `${{ github.sha }}`), but I'm not certain (and the environment variable definitely works).

Reply:
Ignore it then!